wrappers/python/simtk/openmm/app/forcefield.py

```diff
@@ -1674,6 +1674,171 @@ def postprocessSystem(self, sys, data, args):
 
 parsers["NonbondedForce"] = NonbondedGenerator.parseElement
 
+class LennardJonesGenerator(object):
+    """A NBFix generator to construct the L-J force with NBFIX implemented as a lookup table"""
+
+    def __init__(self, forcefield, lj14scale):
+        self.ff = forcefield
+        self.nbfix_types = {}
+        self.lj14scale = lj14scale
+        self.lj_types = ForceField._AtomTypeParameters(forcefield, 'LennardJonesForce', 'Atom', ('sigma', 'epsilon'))
+
+    def registerNBFIX(self, parameters):
+        types = self.ff._findAtomTypes(parameters, 2)
+        if None not in types:
+            type1 = types[0][0]
+            type2 = types[1][0]
+            emin = _convertParameterToNumber(parameters['emin'])
+            rmin = _convertParameterToNumber(parameters['rmin'])
+            self.nbfix_types[(type1, type2)] = [emin, rmin]
+            self.nbfix_types[(type2, type1)] = [emin, rmin]
+
+
+    def registerLennardJones(self, parameters):
+        self.lj_types.registerAtom(parameters)
+
+
+    @staticmethod
+    def parseElement(element, ff):
+        existing = [f for f in ff._forces if isinstance(f, LennardJonesGenerator)]
+        if len(existing) == 0:
+            generator = LennardJonesGenerator(ff, float(element.attrib['lj14scale']))
+            ff.registerGenerator(generator)
+        else:
+            # Multiple <LennardJonesForce> tags were found, probably in different files
+            generator = existing[0]
+
+        for LJ in element.findall('Atom'):
+            generator.registerLennardJones(LJ.attrib)
+        for Nbfix in element.findall('NBFixPair'):
+            generator.registerNBFIX(Nbfix.attrib)
+
+    def createForce(self, sys, data, nonbondedMethod, nonbondedCutoff, args):
+
+        #Ewald and PME will be interpreted as CutoffPeriodic for the CustomNonbondedForce
+
+        # We need a CustomNonbondedForce to implement the NBFIX functionality.
+        # First derive the lookup tables
+        lj_indx_list = [0 for atom in data.atoms]
+        num_lj_types= 0
+        lj_type_list = []
+        type_map = {}
+        for i, atom in enumerate(data.atoms):
+            atype = data.atomType[atom]
+            values = self.lj_types.paramsForType[atype]
+            if lj_indx_list[i]: continue # ljtype has been assigned an index
+            if values in type_map.values() and not atype in [type for key in self.nbfix_types.keys() for type in key]:
```

Comment:
For the second clause, I can add a ```set()``` of nbfix atom types to ```LennardJonesForce``` so then checking membership will be fast. Something like this:

```
def __init__(self, forcefield):
    ...
    self.nbfix_set = set()
```
in ```registerNBFIX```:

 ```
registerNBFIX(self, parameters):
    ....
    self.nbfix_set.update([type1, type2])
```
Then the second clause become:
```atype not in self.nbfix_set:```

If I do ```type_map``` the other way around, mapping values to indices, How will I populate ```acoef``` and ```bcoef```? Right now I loop through the indices and get their values. I can use an ```OrderedDict``` but that will be slow.